test/unit/foreman/http_proxy_test.rb

```diff
@@ -118,16 +118,19 @@ class DummyHTTPAdapter
     end
 
     test 'set @data[:proxy] to proxy' do
-      net_http.request({})
-      assert_equal URI.parse(http_proxy),
-                   net_http.instance_variable_get(:@proxy_address)
+      # stub webmock and the stub our own extension on top of that
+      stub_request(:get, "http://remotehost.xyz/features").to_return(status: 200, body: "", headers: {})
+      load "foreman/http_proxy/net_http_extension.rb"
+      net_http.request(Net::HTTP::Get.new("/features"))
+      assert_equal URI.parse(http_proxy), net_http.instance_variable_get(:@proxy_address)
```

Comment:
This broke jenkins, @timogoebel any idea how to test this? The thing is, webmock shadows our own monkey code which sets the proxy_address variable, I tried to overwrite it but it ends up in "Stack too deep" because its calling its own code. I wonder if this is testable at all.

Reply:
I need to take a look at this before I can answer that, will do next week.

Reply:
Wouldn't you need to use webmock to actually mock a response instead of mocking the library functions?

Reply:
Guys I figured it out, thanks for help. Basically I created an alias and testing the alias which calls the mocked method back. It works as expected now.